### DISCUSSION TITLE
Mongo DB Realm Integration

### DISCUSSION THREAD

**Discussion by SebasScript** [ionic-team/capacitor#3203] [Ideas]
Mongo DB is working on a new database integration for mongo db called realm.
[Mongo DB realm](https://www.mongodb.com/realm)
It is currently in beta and a similar backend as a service product as Firebase. Supporting offline storage, auto sync and other things.
I reached out to mongo db support and suggested an integration for capacitor / cordova (they already got an integration for react native) and they said that they are looking into it as they got multiple similar request.
Having a capacitor integration and being able to use MongoDB with offline storage and auto sync would be an amazing addon to the ecosystem

**Comment by imhoffd** [comment]
This sounds like a plugin request. Care to [make a proposal](https://github.com/capacitor-community/proposals/)?

**Reply by SebasScript** [reply]
Ok made a proposal

**Reply by SebasScript** [reply]
capacitor-community/proposals#44

**Comment by priyankpat** [comment]
https://github.com/capacitor-community/realm I've started the work on it. Completed authentication, looking at the DataModel.

**Reply by SebasScript** [reply]
Nice! And thanks

**Reply by DiegoRealpe** [reply]
Any updates on this @priyankpat ? I saw it was started 4 months ago and having Realm support would make an easy sell for transition out of React Native.

**Comment by jacobg** [comment]
Would this work in local-only mode, without using cloud backend? I want to handle my own syncing to my own cloud service.

**Comment by justinmoon** [comment]
I'm really love a realm integration and it seems like @priyankpat's project would require a ton of work to become usable.
